### PR TITLE
Middleware to check possibility to edit goal/comment and etc

### LINKED ISSUE #1180: Middleware to check possibility to edit goal/comment and etc.
After #1179

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Feature

## Related issues

Resolve #1180

### CODE REVIEW

File: trpc/access/accessCheckers.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+export const accessCheckers = {
+    estimates: {
+        ranges: () => allowed(),
+    },
+    feedback: {
+        create: () => allowed(),
+    },
+    filter: {
+        getById: () => allowed(),
```
Comment (awinogradov) [inline trpc/access/accessCheckers.ts:47]: `getById: allowed`

File: trpc/access/accessCheckers.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+const projectAccessChecker = (session: Session, project: ProjectEntity) => {
+    const { activityId, role } = session.user;
+
+    return project.activityId === activityId || role === 'ADMIN'
+        ? allowed()
+        : notAllowed('No access to update Project');
+};
+
+export const accessCheckers = {
```
Comment (awinogradov) [inline trpc/access/accessCheckers.ts:39]: It can be typed via type of router

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: trpc/access/accessCheckers.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import { Session } from 'next-auth';
+
+import { addCalclulatedGoalsFields } from '../queries/goals';
+
+import { CommentEntity, GoalEntity, ProjectEntity } from './accessEntityGetters';
+
+type AccessCheckerResult = Readonly<{ allowed: true } | { allowed: false; errorMessage: string }>;
+
+export type AccessChecker = (session: Session) => AccessCheckerResult;
```
Comment (awinogradov) [inline trpc/access/accessCheckers.ts:9]: Unused type

**Review by awinogradov — APPROVED** [review]

File: trpc/access/accessCheckers.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+export const commentAccessChecker = (session: Session, comment: CommentEntity) => {
+    const { activityId } = session.user;
+
+    return comment.activityId === activityId ? allowed() : notAllowed('No access to update Comment');
+};
+
+export const projectAccessChecker = (session: Session, project: ProjectEntity) => {
+    const { activityId, role } = session.user;
+
+    return project.activityId === activityId || role === 'ADMIN'
```
Comment (9teen90nine) [inline trpc/access/accessCheckers.ts:33]: We just made _isEditable for projects, check it out https://github.com/taskany-inc/issues/issues/1629

Comment (asabotovich) [reply trpc/access/accessCheckers.ts:33]: thanks

**Pull request merged**